tests/foreman/cli/test_template.py

```diff
@@ -74,6 +80,60 @@ def test_update_template_1(self):
         self.assertEqual(len(result.stderr), 0)
         self.assertEqual(updated_name, result.stdout['name'])
 
+    def test_create_template_with_location(self):
+        """@Test: Check if Template with Location can be created
+
+        @Feature: Template - Create
+
+        @Assert: Template is created and new Location has been assigned
+
+        """
+        try:
+            new_loc = make_location()
+            new_template = make_template({
+                'name': gen_string('alpha', 10),
+                'location-ids': new_loc['id'],
+            })
+        except CLIFactoryError as err:
+            self.fail(err)
+
+        self.assertIn(new_loc['name'], new_template['locations'])
+
+    def test_create_template_locked(self):
+        """@Test: Check that locked Template cannot be created
+
+        @Feature: Template - Create
+
+        @Assert: It is not allowed to create locked Template
+
+        """
+        with self.assertRaises(CLIFactoryError) as ex:
+            make_template({
+                'name': gen_string('alpha', 10),
+                'locked': 'true',
+            })
+        self.assertIn("This template is locked. Please clone it",
+                      ex.exception.message)
```

Comment:
@elyezer Thanks for the inputs, but we cannot use "raw" way as the main task here is actually test factory, so we have to work with make_template whether we want to use it or not. I really like idea to work with localized content and some experience of test automation in that direction, but it is definitely out of scope of current task as it is really huge efforts to create good enough approach to work with localizations (create necessary dictionaries and etc)

@Ichimonji10 I debugged code as more as I can and run necessary statements through hammer and cannot see any reference to error codes.
We have absolutely the same behavior from hammer perspective, for example, for
"Name is too long (maximum is 255 characters)" exception error.

I have tried also to hack 'create_object' method in factory to see return code. And was more successful:):
    if result.return_code != 0:
        logger.debug(result.stderr)  # Show why creation failed.
        raise CLIFactoryError(
            'Failed to create %s with %r data due to:\n%s%d' % (
                cli_object.__name__,
                options,
                _format_error_msg(result.stderr),
                result.return_code
            )
        )

Received '65' error code for both 'template locked' and 'name too long' exceptions, so no difference.

Of course, there are a lot of different error codes, but that particular example shows that we can have unambiguous results and anyway we need to move to really another level if we want to have verification like this